arrow-libs/core/arrow-core/src/test/kotlin/arrow/core/EitherTest.kt

```diff
@@ -162,14 +229,78 @@ class EitherTest : UnitSpec() {
       }
     }
 
+    "map should alter right instance only" {
+      forAllSmallInt { a: Int, b: Int ->
+        val right: Either<Int, Int> = Right(a)
+        val left: Either<Int, Int> = Left(b)
+
+        right.map { it + 1 } == Right(a + 1) && left.map { it + 1 } == left
+      }
+    }
+
     "mapLeft should alter left instance only" {
-      forAll(Gen.intSmall(), Gen.intSmall()) { a: Int, b: Int ->
+      forAllSmallInt { a: Int, b: Int ->
         val right: Either<Int, Int> = Right(a)
         val left: Either<Int, Int> = Left(b)
+
         right.mapLeft { it + 1 } == right && left.mapLeft { it + 1 } == Left(b + 1)
       }
     }
 
+    "bimap should alter left or right instance accordingly" {
+      forAllSmallInt { a: Int, b: Int ->
+        val right: Either<Int, Int> = Right(a)
+        val left: Either<Int, Int> = Left(b)
+
+        right.bimap({ it + 2 }, { it + 1 }) == Right(a + 1) &&
+          left.bimap({ it + 2 }, { it + 1 }) == Left(b + 2)
+      }
+    }
+
+    "replicate should return Right(empty list) when n <= 0" {
+      forAll(
+        Gen.oneOf(Gen.negativeIntegers(), Gen.constant(0)),
+        Gen.int()
+      ) { n: Int, a: Int ->
+        val expected: Either<Int, List<Int>> = Right(emptyList())
+
+        Right(a).replicate(n) == expected &&
+          Left(a).replicate(n) == expected
+      }
+    }
+
+    "replicate should return Right(list of repeated value size n) when Right and n is positive" {
+      forAll(
+        Gen.intSmall().filter { it > 0 },
+        Gen.int()
+      ) { n: Int, a: Int ->
+        Right(a).replicate(n) == Right(List(n) { a }) &&
+          Left(a).replicate(n) == Left(a)
+      }
+    }
+
+    "traverse should return list of Right when Right and empty list when Left" {
+      forAll(
+        Gen.int(),
+        Gen.int(),
+        Gen.int()
+      ) { a: Int, b: Int, c: Int ->
```

Comment:
Can test with `Gen.list`, but I thought that showing list of `Right` is more explicit.